xrtpy/util/xrt_remove_lightleak.py

```diff
@@ -0,0 +1,275 @@
+        leak_fn = sl_phase_dict[sl_phase]
+    except KeyError:
+        print(f"No leak image available for phase {sl_phase}")
+        if verbose:
+            print(f"Filter pair used: {sfpair}")
+        print("Output data are identical with the input")
+        return in_map
+
+    leak_map = Map(dir_leak / leak_fn)
+
+    leak_image = leak_map.data
+    if in_map.meta["naxis1"] == 2048:
+        # assumes leak_image is 1024 x 1024 (I think)
+        leak_image2 = rebin(leak_image, (2048, 2048)) * 0.25 * kfact
+    else:
+        leak_image2 = leak_image * kfact
+
+    out_image = in_map.data - leak_image2
+
+    hist_entry = f"Light leak subtraction: DONE with {leak_fn} and kfactor:{kfact}"
+    out_meta = in_map.meta
+    out_meta["history"] += f"\n{hist_entry}"
+
+    if verbose:
+        print(f"appended to the history : {hist_entry}")
+
+    return Map((out_image, out_meta))
+
+
+def check_sl_phase(date_obs):
+    """
+    Get the stray light phase number
+
+    Parameters:
+    -----------
+    obs_date : `str`
+        Observation date string formatted in ISO format: YYYY-MM-DDTHH:mm
+        where YYYY is the year, MM is the month number (zero-padded), DD is
+        the day of the month (zero-padded), HH is the hour and mm is the
+        minute (note T separator between date and time; may also include
+        seconds). This is the standard format for the value of date_obs in the
+        FITS headers of XRT images.
+
+    Returns:
+    --------
+    phase no. : integer
+        The phase of the light leak as follows:
+
+    phase 0 : 23-Oct-2006 10:00
+    phase 1 :  9-May-2012 12:00
+    phase 2 : 14-Jun-2015 12:30
+    phase 3 : 27-May-2017 11:00
+    phase 4 : 29-May-2018 00:00
+    phase 5 :  8-Jun-2022 12:40
+    phase 6 :  5-May-2023  4:39
+    """
+
+    time_p1 = datetime.strptime("09-May-2012 12:00", "%d-%b-%Y %H:%M")
+    time_p2 = datetime.strptime("14-Jun-2015 12:30", "%d-%b-%Y %H:%M")
+    time_p3 = datetime.strptime("27-May-2017 11:00", "%d-%b-%Y %H:%M")
+    time_p4 = datetime.strptime("29-May-2018 00:00", "%d-%b-%Y %H:%M")
+    time_p5 = datetime.strptime("08-Jun-2022 12:40", "%d-%b-%Y %H:%M")
+    time_p6 = datetime.strptime("05-May-2023 04:39", "%d-%b-%Y %H:%M")
+
+    intime = datetime.fromisoformat(date_obs)
+    if intime >= time_p6:
+        phase = 6
+    elif intime >= time_p5:
+        phase = 5
+    elif intime >= time_p4:
+        phase = 4
+    elif intime >= time_p3:
+        phase = 3
+    elif intime >= time_p2:
+        phase = 2
+    elif intime >= time_p1:
+        phase = 1
+    else:
+        phase = 0
+
+    return phase
+
+
+def my_rebin(image, newdims):
+    """Rebin an array to a new shape"""
+
+    newimage = np.zeros(newdims)
+    origdims = image.shape
+    ifac = newdims[0] // origdims[0]
+    jfac = newdims[1] // origdims[1]
+    for i in range(ifac):
+        for j in range(jfac):
+            newimage[i::ifac, j::jfac] = image[:]
+    return newimage
+
+
+def rebin(image, newshape):
```

Comment:
```suggestion
def _rebin(image, newshape):
```
Perhaps rename this to `_rebin` to indicate that it is a private function?  We'd need to change it above too.

Reply:
We could certainly make it a private function, though I don't see it as necessary. It's my understanding that one should make functions private if they're subject to change with no notice. The idea is that you don't want users to depend on the functions future existence. In this case it's such a simple function that it's unlikely to change. Granted a utility function like this one should probably live elsewhere.